File: logic.js
 function get_days_of_power(k){


               return  function get_days_of_power(R1, D1, D2 ){
                    var ans = [];
                    // loop through the range of days for power

                for (let i = D1; i <= D2; i++) {
                    ans.push(i);
                }
                // check the length of the array which is the total number of the account days range

                let dayOfPower1 = ans.length
                let sumOfdayRate = dayOfPower1 * R1;
                if(k > sumOfdayRate){
                    return k -sumOfdayRate
                }


                return function get_days_of_power(R2, D2, D3 ){
                    var ans = [];
                    // loop through the range of days for power
                for (let i = D2; i <= D3; i++) {
                    ans.push(i);
                }

                // check the length of the array which is the total number of the account days range

                let dayOfPower2 = ans.length
                let sumOfdayRate = dayOfPower2 * R2;
                if(k > sumOfdayRate){
                    return k -sumOfdayRate
                }
                
                return  function get_days_of_power(R3, D3, D4 ){
                    var ans = [];
                     // loop through the range of days for power
                for (let i = D3; i <= D4; i++) {
                    ans.push(i);
                }
                // check the length of the array which is the total number of the account days range

                let dayOfPower3 = ans.length
                let sumOfdayRate =  dayOfPower3 * R3;
                if(k > sumOfdayRate){
                    return k - sumOfdayRate
                }
                return dayOfPower1+dayOfPower2+dayOfPower3
                
            }

        }

    }
}
    
    
    
    
    
   







     
  
   
   
get_days_of_power(3,6)

File: logic.py
def get_days_of_power_d1(D1,D2):
    ans = []
    for i in range(D1,D2):
        ans.append(i) 
        print(len(ans))


def get_days_of_power_d2(D2,D3):
    ans = []
    for i in range(D2,D3):
        ans.append(i) 
        print(len(ans))


def get_days_of_power_d3(D3,D4):
    ans = []
    for i in range(D3,D4):
        ans.append(i) 
        print(len(ans))
       

get_days_of_power_d1(3,7)   


def get_days_of_power(R1,get_days_of_power_D1):
    d1 = get_days_of_power_D1()
    

    dayOfPower1 = d1 * R1
    print(dayOfPower1)


get_days_of_power(1000,3)    





